### ISSUE TITLE
Add more information and links about contributing

### ISSUE THREAD

**Issue by kevinrobinson** [studentinsights/studentinsights#235]
Here are some great models for how other projects help folks contribute, especially first-time contributors.  We should adapt these for this project and add them to the `contributing` section of the README.

- [Write a contribution guide documenting the design decisions · Issue #6335 · facebook/react](https://github.com/facebook/react/issues/6335)
- [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github)

**Comment by dazzaji** [comment]
Working on this tonight, following a GitHub standard flow, described [here](https://help.github.com/articles/setting-guidelines-for-repository-contributors).  The root-level file named CONTRIBUTING.md, can/should be linked to from appropriate places in the more general project README file.  

This is especially handy because when a repo contains a file named CONTRIBUTING.md then when people add issues GitHub will automatically trigger display of a notice along these lines: 
![contributing-guidelines](https://cloud.githubusercontent.com/assets/2357755/9638557/6b3ac6c2-5175-11e5-8fd3-dd6ad44db9b1.jpg)

According to GitHub: 
> In the new file, add contribution guidelines. These could include:
> * Steps for creating good issues or pull requests.
> * Links to external documentation, mailing lists, or a code of conduct.
> * Community and behavioral expectations.

Here goes...

**Comment by alexsoble** [comment]
Very cool, did not know that bit about the flash notice @dazzaji! A nice bit of sugar to help the community contribution process flow smoother! :rainbow:

**Comment by alexsoble** [comment]
Hey look, here it is!

![contributing](https://cloud.githubusercontent.com/assets/3209501/14157974/43413c96-f693-11e5-92d8-1f5b0e3eb701.png)

**Comment by dazzaji** [comment]
Looking good!  So, is this ready to close or are there more things to do?

**Comment by kevinrobinson** [comment]
@dazzaji This is an awesome improvement, thank you!  It'd be great to add the egghead links to how to do the GitHub flow (since walking through this has come up several times over the last few weeks at Code for Boston).

For the sake of good issue hygeine, let's cut the vaguer bits about context and high-level design decisions for now.  But for the thought trail, things that have come up a few times already are why are we defining styles directly in React  (eg., https://speakerdeck.com/vjeux/react-css-in-js as background), or how to make me JS modules with each file stuffed into `window.shared` at the moment.

**Comment by dazzaji** [comment]
I'm not exactly sure what you are suggesting be cut. Can you go ahead and make this change and then close this issue? Thanks.

**Comment by kevinrobinson** [comment]
@dazzaji ahh sorry for being confusing.  I lumped a few things into this issue initially, just to capture it and I can see why that is totally lost without mind reading skills. :)  I'm suggesting part of my issue be cut, not your awesome contribution

I split the remaining bit out to https://github.com/studentinsights/studentinsights/issues/251, and clarified that that bit is about documentation high-level design decisions, since those have come up a few times at Code for Boston and it's good to externalize.

I also added a commit here https://github.com/studentinsights/studentinsights/pull/248/commits/89971854bd8fd14588b8358d5485681d09fbdbc7 to the a more walkthrough-style tutorial on the GitHub workflow so we can :closed_book: this sucker out.  Thanks again @dazzaji and for speaking up about how I could help clarify. :)
